backend/backend/core/tools.py

```diff
@@ -1,4 +1,5 @@
 import os
+import re
 
 from pydantic import BaseModel, Field
 from typing import Optional, Type
```

Comment:
The import of langchain_core.tools is not used in this code snippet. Consider removing unused imports to declutter the code.